packaging_automation/common_tool_methods.py

```diff
@@ -0,0 +1,286 @@
+import os
+import re
+import subprocess
+from datetime import datetime
+from typing import Dict, List
+from typing import Tuple
+
+import pathlib2
+from git import Repo
+from github import Repository, PullRequest, Commit
+from jinja2 import Environment, FileSystemLoader
+
+from .common_validations import (is_tag, is_version)
+
+BASE_GIT_PATH = pathlib2.Path(__file__).parents[1]
+PATCH_VERSION_MATCH_FROM_MINOR_SUFFIX = "\.\d{1,3}"
+
+# When using GitPython library Repo objects should be closed to be able to delete cloned sources
+# referenced by Repo objects. References are stored below array to be able to close all resources after code execution
+referenced_repos:List[Repo] = []
+
+
+def get_new_repo(working_dir: str) -> Repo:
+    repo = Repo(working_dir)
+    referenced_repos.append(repo)
+    return repo
+
+
+def release_all_repos():
+    for repo in referenced_repos:
+        repo.close()
+
+
+def get_spec_file_name(project_name: str) -> str:
+    return f"{project_name}.spec"
+
+
+def get_project_version_from_tag_name(tag_name: is_tag(str)) -> str:
+    return tag_name[1:]
+
+
+def get_template_environment(template_dir: str) -> Environment:
+    file_loader = FileSystemLoader(template_dir)
+    env = Environment(loader=file_loader)
+    return env
+
+
+def find_nth_occurrence_position(subject_string: str, search_string: str, n) -> int:
+    start = subject_string.find(search_string)
+
+    while start >= 0 and n > 1:
+        start = subject_string.find(search_string, start + 1)
+        n -= 1
+    return start
+
+
+def find_nth_matching_line_and_line_number(subject_string: str, regex_pattern: str, n: int) -> Tuple[int, str]:
+    """Takes a subject string, regex param and the search index as parameter and returns line number of found match.
+    If not found returns -1"""
+    lines = subject_string.splitlines()
+    counter = 0
+    for line_number, line in enumerate(lines):
+        if re.match(regex_pattern, line):
+            counter = counter + 1
+        if counter == n:
+            return line_number, lines[line_number]
+    return -1, ""
+
+
+def remove_text_with_parenthesis(param: str) -> str:
+    """Removes texts within parenthesis i.e. outside parenthesis(inside parenthesis)-> outside parenthesis """
+    return re.sub(r"[(\[].*?[)\]]", "", param)
+
+
+def run(command, *args, **kwargs):
+    result = subprocess.run(command, *args, check=True, shell=True, **kwargs)
+    return result
+
+
+def cherry_pick_prs(prs: List[PullRequest.PullRequest]):
+    for pr in prs:
+        commits = pr.get_commits()
+        for single_commit in commits:
+            if not is_merge_commit(single_commit):
+                cp_result = run(f"git cherry-pick {single_commit.commit.sha}")
```

Comment:
We usually use `-x` parameter to cherry-pick the commits:

```
       -x
           When recording the commit, append a line that says "(cherry picked from commit ...)" to the original
           commit message in order to indicate which commit this change was cherry-picked from. ...
```

```suggestion
                cp_result = run(f"git cherry-pick -x {single_commit.commit.sha}")
```

Reply:
Fixed